image/image_test.go

```diff
@@ -0,0 +1,479 @@
+				So(err, ShouldBeNil)
+				So(m.Id, ShouldResemble, "042e216a-7822-4fa0-a3d6-e3f5248ffc35")
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkResponseBase(mockdphttpCli, http.MethodPost, "/images")
+				payload, err := ioutil.ReadAll(mockdphttpCli.DoCalls()[0].Req.Body)
+				So(err, ShouldBeNil)
+				So(payload, ShouldResemble, expectedPayload)
+			})
+		})
+	})
+
+	Convey("given a 404 status is returned", t, func() {
+		mockdphttpCli := createHTTPClientMock(http.StatusNotFound, []byte("wrong!"))
+		cli := Client{cli: mockdphttpCli, url: "http://localhost:8080"}
+		expectedPayload, err := json.Marshal(newImage)
+		So(err, ShouldBeNil)
+
+		Convey("when PostInstanceDimensions is called", func() {
+			_, err := cli.PostImage(ctx, newImage)
+
+			Convey("then the expected error is returned", func() {
+				So(err.Error(), ShouldResemble, errors.Errorf("invalid response: 404 from image api: http://localhost:8080/images, body: wrong!").Error())
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkResponseBase(mockdphttpCli, http.MethodPost, "/images")
+				payload, err := ioutil.ReadAll(mockdphttpCli.DoCalls()[0].Req.Body)
+				So(err, ShouldBeNil)
+				So(payload, ShouldResemble, expectedPayload)
+			})
+		})
+	})
+}
+
+func TestClient_GetImage(t *testing.T) {
+	Convey("given a 200 status is returned", t, func() {
+		searchResp, err := ioutil.ReadFile("./response_mocks/image.json")
+		So(err, ShouldBeNil)
+
+		mockdphttpCli := createHTTPClientMock(http.StatusOK, searchResp)
+		cli := Client{cli: mockdphttpCli, url: "http://localhost:8080"}
+
+		Convey("when GetImages is called", func() {
+			m, err := cli.GetImage(ctx, "123")
+
+			Convey("a positive response is returned", func() {
+				So(err, ShouldBeNil)
+				So(m.Id, ShouldResemble, "042e216a-7822-4fa0-a3d6-e3f5248ffc35")
+
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkResponseBase(mockdphttpCli, http.MethodGet, "/images/123")
+			})
+		})
+	})
+
+	Convey("given a 404 status is returned", t, func() {
+		mockdphttpCli := &dphttp.ClienterMock{
+			DoFunc: func(ctx context.Context, req *http.Request) (*http.Response, error) {
+				return &http.Response{
+					StatusCode: http.StatusNotFound,
+					Body:       ioutil.NopCloser(bytes.NewReader([]byte("resource not found"))),
+				}, nil
+			},
+		}
+
+		cli := Client{cli: mockdphttpCli, url: "http://localhost:8080"}
+
+		Convey("when GetInstanceDimensionsBytes is called", func() {
+			_, err := cli.GetImage(ctx, "123")
+
+			Convey("then the expected error is returned", func() {
+				So(err.Error(), ShouldResemble, errors.Errorf("invalid response: 404 from image api: http://localhost:8080/images/123, body: resource not found").Error())
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkResponseBase(mockdphttpCli, http.MethodGet, "/images/123")
+			})
+		})
+	})
+}
+
+func TestClient_PutImage(t *testing.T) {
```

Comment:
If we decide to use the iota enum, here we could add a test with an invalid state